lib/remap-functions/src/parse_common_log.rs

```diff
@@ -0,0 +1,243 @@
+                ),
+            );
+        }
+
+        if let Some(request_line) = captures
+            .name("request_line")
+            .map(|capture| capture.as_str())
+        {
+            log.insert(
+                "request_line".into(),
+                Value::Bytes(request_line.to_owned().into()),
+            );
+        }
+
+        if let Some(request_method) = captures
+            .name("request_method")
+            .map(|capture| capture.as_str())
+        {
+            log.insert(
+                "request_method".into(),
+                Value::Bytes(request_method.to_owned().into()),
+            );
+        }
+
+        if let Some(request_path) = captures
+            .name("request_path")
+            .map(|capture| capture.as_str())
+        {
+            log.insert(
+                "request_path".into(),
+                Value::Bytes(request_path.to_owned().into()),
+            );
+        }
+
+        if let Some(request_protocol) = captures
+            .name("request_protocol")
+            .map(|capture| capture.as_str())
+        {
+            log.insert(
+                "request_protocol".into(),
+                Value::Bytes(request_protocol.to_owned().into()),
+            );
+        }
+
+        if let Some(status_code) = captures.name("status_code").map(|capture| capture.as_str()) {
+            log.insert(
+                "status_code".into(),
+                Value::Integer(
+                    status_code
+                        .parse()
+                        .map_err(|_| "failed parsing status code.")?,
+                ),
+            );
+        }
+
+        if let Some(content_length) = captures
+            .name("content_length")
+            .map(|capture| capture.as_str())
+        {
+            log.insert(
+                "content_length".into(),
+                Value::Integer(
+                    content_length
+                        .parse()
+                        .map_err(|_| "failed parsing content length.")?,
+                ),
+            );
+        }
+
+        Ok(log.into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .fallible_unless(value::Kind::Bytes)
+            .with_constraint(value::Kind::Map)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use shared::btreemap;
+
+    test_function![
+        parse_common_log => ParseCommonLog;
+
+        log_line_valid {
+            args: func_args![value: r#"127.0.0.1 bob frank [10/Oct/2000:13:55:36 -0700] "GET /apache_pb.gif HTTP/1.0" 200 2326"#],
+            want: Ok(btreemap! {
+                "remote_host" => "127.0.0.1",
+                "remote_logname" => "bob",
+                "auth_user" => "frank",
+                "timestamp" => Value::Timestamp(DateTime::parse_from_rfc3339("2000-10-10T20:55:36Z").unwrap().into()),
+                "request_line" => "GET /apache_pb.gif HTTP/1.0",
```

Comment:
I think still `message` is a better name for this, right?

Reply:
Hm, both W3C and Apache call it request line:

https://www.w3.org/Daemon/User/Config/Logging.html#common-logfile-format

> The request line exactly as it came from the client.

https://httpd.apache.org/docs/1.3/logs.html#common

> The request line from the client is given in double quotes.

As it contains metadata and not any actual message, I'm not sure message is a good term here.

Do we use "message" in this context in other places?

Reply:
Yes, we typically use `timestamp`, `message`, and `host`. It's all very loose right now, Vector doesn't care what you call the fields. We will, at some point in the future, allow users to choose a schema and Vector will use that for field names.

Reply:
They're called

- `host`
- `identity`
- `user`
- `timestamp`
- `message`
- `method`
- `path`
- `protocol`
- `status`
- `size`

now.